src/main/java/org/codehaus/plexus/archiver/jar/JdkManifestFactory.java

```diff
@@ -33,25 +33,33 @@
     public static java.util.jar.Manifest getDefaultManifest()
         throws ArchiverException
     {
-        final java.util.jar.Manifest defaultManifest = new java.util.jar.Manifest();
-        defaultManifest.getMainAttributes().putValue( "Manifest-Version", "1.0" );
+        try
+        {
+            final java.util.jar.Manifest defaultManifest = new java.util.jar.Manifest();
+            defaultManifest.getMainAttributes().putValue( "Manifest-Version", "1.0" );
 
-        String createdBy = "Plexus Archiver";
+            String createdBy = "Plexus Archiver";
 
-        InputStream inputStream = JdkManifestFactory.class.getResourceAsStream( "/META-INF/"
-            + "maven/org.codehaus.plexus/plexus-archiver/pom.properties" );
-        Properties properties = PropertyUtils.loadProperties( inputStream );
-        if ( properties != null )
-        {
-            String plexusArchiverVersion = properties.getProperty( "version" );
-            if ( plexusArchiverVersion != null )
+            final Properties properties = PropertyUtils.loadProperties( JdkManifestFactory.class.getResourceAsStream(
+                "/META-INF/maven/org.codehaus.plexus/plexus-archiver/pom.properties" ) );
+
+            if ( properties != null )
             {
-                createdBy += " " + plexusArchiverVersion;
+                String plexusArchiverVersion = properties.getProperty( "version" );
+                if ( plexusArchiverVersion != null )
+                {
+                    createdBy += " " + plexusArchiverVersion;
+                }
             }
-        }
-        defaultManifest.getMainAttributes().putValue( "Created-By", createdBy );
 
-        return defaultManifest;
+            defaultManifest.getMainAttributes().putValue( "Created-By", createdBy );
+
+            return defaultManifest;
+        }
+        catch ( final IOException e )
+        {
```

Comment:
Am 2016-05-27 um 02:51 schrieb Christian Schulte:
>>
>> -        return defaultManifest;
>> +            defaultManifest.getMainAttributes().putValue( "Created-By", createdBy );
>> +
>> +            return defaultManifest;
>> +        }
>> +        catch ( final IOException e )
>> +        {
>
> Am 05/27/16 um 00:57 schrieb Michael Osipov:
>> In
>> src/main/java/org/codehaus/plexus/archiver/jar/JdkManifestFactory.java
>> <https://github.com/codehaus-plexus/plexus-archiver/pull/41#discussion_r64835114>:
>>
>>>
>>> -        return defaultManifest;
>>> +            defaultManifest.getMainAttributes().putValue( "Created-By", createdBy );
>>> +
>>> +            return defaultManifest;
>>> +        }
>>> +        catch ( final IOException e )
>>> +        {
>>
>> I consider the swallow here
>> <https://github.com/codehaus-plexus/plexus-utils/commit/ba1c194dc0d57b49ffb2220b7571e68caf09f18b>
>> was intentional. The local was actually borrowed from other spots where
>> the version is omitted of the |pom.properties| could not be read.
>>
>
> Just restored that behaviour. I even considered throwing an
> AssertionError if the properties cannot be read. Omitting the version

I am not convinced that `AssertionError` is the right thing because 
`Errors` are reverved for the VM. Please search for `pom.properties` in 
`maven-core` (`DefaultRepositorySystemSessionFactory` and 
`DefaultRuntimeInformation`). You will see the basic ida it was taken from.